logging/google/cloud/logging/logger.py

```diff
@@ -473,6 +517,8 @@ def commit(self, client=None):
                 info = {'protoPayload': MessageToDict(entry)}
             else:
                 raise ValueError('Unknown entry type: %s' % (entry_type,))
+            if resource is not None:
+                info['resource'] = resource._to_dict()
```

Comment:
ISTM that unless a user **explicitly** sends `None`, this branch will always execute (since the default is `_GLOBAL_RESOURCE`)